HW_6/HW_5a.py

```diff
@@ -0,0 +1,144 @@
+
+import datetime
+import random
+import datefinder
+import HW_3_updated as normalization
+import sys, os
+
+sys.path.append('C:\\Users\\Kateryna_Kulmatytska\\PycharmProjects\\Python_HW\\HW_6')
+
+
+class Article:
+    def __init__(self, header, text):
+        self.header = header
+        self.text = text
+
+    def print_to_file(self, formatted_text, file_name):
+        with open(file_name, "a") as file:
+            file.write(formatted_text)
+
+
+class News(Article):
+    def __init__(self, header, text, location):
+        Article.__init__(self, header=header, text=text)
+        self.location = location
+
+    def format_text(self, header, text, location):
+        return f"{header}--------------" \
+               f"\n{text}\n" \
+               f"{location},{datetime.datetime.today().strftime('%d/%m/%Y %H.%M')}" \
+               f"\n-------------------\n\n"
+
+
+class PrivatAdv(Article):
+    def __init__(self, header, text, expiration_date):
+        Article.__init__(self, header=header, text=text)
+        self.expiration_date = expiration_date
+
+    def format_text(self, header, text, expiration_date):
+        return f"{header}----------" \
+               f"\n{text}\n" \
+               f"Actual until: {expiration_date}, {(expiration_date - datetime.date.today()).days} days left." \
+               f"\n-----------------------\n\n"
+
+
+class Consider(Article):
+    def __init__(self, header, text, score):
+        Article.__init__(self, header=header, text=text)
+        self.score = score
+
+    def format_text(self, header, text, score):
+        return f"{header}-------------- " \
+               f"\n{text}\n" \
+               f"Probability of truth:{score}%" \
+               f"\n------------------------------\n\n"
+
+
+def write_article():
+    input_versions = ('News', 'PrivatAdv', 'Consider')
+    header = input('Enter the topic you are interested in:\nNews\nPrivatAdv\nConsider\n:')
+
+    while header not in input_versions:
+        print('Please select correct topic:')
+        header = input('News\nPrivatAdv\nConsider\n:')
+
+    print("Type your text here. Press Enter twice to exit from text adding mode")
+    input_text = ''
+    while True:
+        new_line = input()
+        if new_line != '':
+            input_text += new_line + '\n'
+        else:
+            input_text = input_text[:-1]
+            break
+    text = normalization.normalization(input_text)
+    if header == 'News':
+        if len(text) == 0:
+            user_input = input("Enter the path of your file: ")
+            if os.path.exists(user_input):
+                input_file = open(user_input, 'r+')
```

Comment:
If an input file exists, you open it on line 79, and then close it on line 89. So a user file isn't processed in any case.